codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/codegen/knowledge/GoPointableIndex.java

```diff
@@ -173,7 +174,12 @@ private boolean isMemberPointable(MemberShape member, Shape targetShape) {
             return false;
         }
 
-        if (NUMBER_SHAPE.contains(targetShape.getType()) && !member.hasTrait(RequiredTrait.class) && !model.expectShape(member.getContainer()).isMapShape()) {
+        //if membershape is required return it is not pointable
+        if (member.hasTrait(RequiredTrait.class) && !STRICT_POINTER_TYPE.contains(targetShape.getType())) {
+            return false;
+        }
+
+        if (KNOWN_POINTER_TYPE.contains(targetShape.getType()) && !model.expectShape(member.getContainer()).isMapShape()) {
```

Comment:
I think this is the right path. There SHOULD NOT be any case where we need `*` as value for map.